### ISSUE TITLE
Shealth crash w/logs

### ISSUE THREAD

**Issue by ShaDisNX255** [adil192/SamsungAppsPatcher#10]
Hello,

Sorry to bother you, would it be too much bother if you can check out these logs of shealth crashing
Latest one in your download page
Version 6.22.1.053
I used the -b crash flag in logcat, let me know if you need me to capture all logs

Log: Shealth https://pastebin.com/YuqRFwVC

**Comment by adil192** [comment]
```
08-12 07:43:50.003 11829 11829 E AndroidRuntime: FATAL EXCEPTION: main
08-12 07:43:50.003 11829 11829 E AndroidRuntime: Process: com.sec.android.app.shealth:remote, PID: 11829
08-12 07:43:50.003 11829 11829 E AndroidRuntime: java.lang.NoClassDefFoundError: Failed resolution of: Lcom/samsung/android/app/shealth/util/SignatureChecker;
08-12 07:43:50.003 11829 11829 E AndroidRuntime: 	at com.samsung.android.app.shealth.util.k0.<clinit>(SignatureChecker.java:8)
```

Logs would imply I've made some kind of error in `shealth_custom_cert.patch`, but I've just installed shealth on my Pixel and it doesn't crash before/after logging in.

Could you try uninstalling and reinstalling / clearing the app's data? Make sure it's synced with your Samsung account first

Also can you share your phone manufacturer (Samsung?) and Android version?

**Comment by ShaDisNX255** [comment]
I can give it a try, sure. Yeah, I've attempted to do everything you suggested and still kept crashing. At first I thought that having it sync with Samsung cloud was the problem but no, it crashed without synching too. My phone is a Galaxy A52s 5G (SM-A528B), though I'm running S21FE's system to get extra features. Your previous SHealth app does work though, just confirming.

**Comment by adil192** [comment]
@ShaDisNX255 Please try https://adil.hanney.org/SamsungAppsPatcher/dl/shealth.issue10.apk

**Comment by ShaDisNX255** [comment]
> > At first I thought that having it sync with Samsung cloud was the problem but no, it crashed without synching too.
> 
> What actions do you take before it crashes? Or does it crash as soon as you open it?

No, it doesn't crash as soon as it opens. It crashes once you leave the main screen. You can log in just fine and be on the app's main screen. As soon as you try to switch to another screen or even minimize it, it would crash.

I've now tested your new version here:

> @ShaDisNX255 Please try https://adil.hanney.org/SamsungAppsPatcher/dl/shealth.issue10.apk

No more crashing. Thank you very much :)
